src/android/com/totalpave/cordova/inset/Inset.java

```diff
@@ -0,0 +1,311 @@
+/*
+   Copyright 2019 Total Pave Inc.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package com.totalpave.cordova.inset;
+
+import android.content.Context;
+import android.os.Build;
+import android.view.RoundedCorner;
+import android.view.WindowInsets;
+import androidx.core.view.ViewCompat;
+import androidx.core.view.WindowInsetsCompat;
+import org.apache.cordova.CallbackContext;
+import org.apache.cordova.CordovaPlugin;
+import org.apache.cordova.PluginResult.Status;
+import org.apache.cordova.PluginResult;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.json.JSONException;
+import java.lang.NumberFormatException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.UUID;
+
+public class Inset extends CordovaPlugin {
+    public static final int DEFAULT_INSET_MASK = WindowInsetsCompat.Type.displayCutout() | WindowInsetsCompat.Type.systemBars();
+    public static final boolean DEFAULT_INCLUDE_ROUNDED_CORNERS = true;
+
+    public static class WebviewMask {
+        private WebviewMask() {}
+
+        public static final int CAPTION_BAR                 = 1;
+        public static final int DISPLAY_CUTOUT              = 1 << 1;
+        public static final int IME                         = 1 << 2;
+        public static final int MANDATORY_SYSTEM_GESTURES   = 1 << 3;
+        public static final int NAVIGATION_BARS             = 1 << 4;
+        public static final int STATUS_BARS                 = 1 << 5;
+        public static final int SYSTEM_BARS                 = 1 << 6;
+        public static final int SYSTEM_GESTURES             = 1 << 7;
+        public static final int TAPPABLE_ELEMENT            = 1 << 8;
```

Comment:
Is this any different from simple assignment? I mean if you wanted to add another option in the middle, you'd need to update everything below it no?

Reply:
There is absolutely no reason to add a line in the middle. These values are replicated and are exclusive to the plugin. They are mapped to the android's sdk constants. This decouples our code from the android's sdk, which doesn't document any explicit constant value for their inset types.